src/pymap/settings.py

```
@@ -356,44 +386,96 @@ def check_log_directory() -> None:
+        raise
 
 # Set the same secret key for debug and testing
-if DEBUG:
+if DEBUG or TESTING:
     SECRET_KEY = "!!DEBUG_KEY!!"
 
+# Call the check_log_directory function during startup
+try:
+    PYMAP_LOGDIR = check_log_directory()
+except (FileNotFoundError, PermissionError) as e:
+    raise ImproperlyConfigured(f"Error checking log directory: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)